service/src/main/java/com/derpgroup/derpwizard/dao/impl/H2EmbeddedAccountLinkingDAO.java

```diff
@@ -0,0 +1,407 @@
+    } catch (SQLException e) {
+      e.printStackTrace();
+    }
+    return null;
+  }
+
+  @Override
+  public String getUserIdByMappingToken(String token) {
+
+    ResultSet response;
+    String linkingTokenRetrieve = "SELECT TOP 1 userId FROM LinkingToken WHERE token = ? ORDER BY dateCreated DESC";
+    try {
+      ArrayList<String> parameters = new ArrayList<String>();
+      parameters.add(token);
+      response = executeQuery(linkingTokenRetrieve, parameters);
+      if(response == null || !response.next()){
+        return null;
+      }
+      return response.getString("userId");
+    } catch (SQLException e) {
+      e.printStackTrace();
+    }
+    return null;
+  }
+
+  @Override
+  public void expireMappingToken(String token) {
+    String linkingTokenDelete = "DELETE FROM LinkingToken WHERE token = ?;";
+
+//    try {
+      ArrayList<String> parameters = new ArrayList<String>();
+      parameters.add(token);
+      executeStatement(linkingTokenDelete, parameters);
+    /*} catch (SQLException e) {
+      e.printStackTrace();
+    }*/
+  }
+
+  @Override
+  public String generateAuthToken(String userId) {
+
+    String accessTokenCreate = "INSERT INTO Authorization(userId) VALUES(?);";
+
+//    try {
+      ArrayList<String> parameters = new ArrayList<String>();
+      parameters.add(userId);
+      executeStatement(accessTokenCreate, parameters);
+    /*} catch (SQLException e) {
+      e.printStackTrace();
+      return null;
+    }*/
+
+    ResultSet response;
+    String accessTokenRetrieve = "SELECT TOP 1 token FROM Authorization WHERE userId = ? ORDER BY dateCreated DESC";
+    try {
+      ArrayList<String> retrievalParameters = new ArrayList<String>();
+      retrievalParameters.add(userId);
+      response = executeQuery(accessTokenRetrieve, retrievalParameters);
+      response.first();
+      return response.getString("token");
+    } catch (SQLException e) {
+      e.printStackTrace();
+    }
+    return null;
+  }
+
+  @Override
+  public String getUserIdByAuthToken(String token) {
+
+    ResultSet response;
+    String accessTokenRetrieve = "SELECT TOP 1 userId FROM Authorization WHERE token = ? ORDER BY dateCreated DESC";
+    try {
+      ArrayList<String> parameters = new ArrayList<String>();
+      parameters.add(token);
+      response = executeQuery(accessTokenRetrieve, parameters);
+      if(!response.next()){
+        return null;
+      }
+      return response.getString("userId");
+    } catch (SQLException e) {
+      e.printStackTrace();
+    }
+    return null;
+  }
+
+  @Override
+  public void expireGrantedToken(String token) {
+    String accessTokenDelete = "DELETE FROM Authorization WHERE token = ?;";
+
+    ArrayList<String> parameters = new ArrayList<String>();
+    parameters.add(token);
+    executeStatement(accessTokenDelete, parameters);
+  }
+  
+  protected PreparedStatement prepStatement(PreparedStatement statement, ArrayList<? extends Object> parameters) throws SQLException{
```

Comment:
Should this instead be?
List < Object > parameters

That's a legit question, I'm not sure what the difference would be here